src/Processors/Merges/Algorithms/SummingSortedAlgorithm.cpp

```diff
@@ -696,6 +696,12 @@ void SummingSortedAlgorithm::SummingMergedData::addRowImpl(ColumnRawPtrs & raw_c
         }
         else
         {
+            if (!def.arena)
+            {
+                def.arena = std::make_unique<Arena>();
+                def.arena_size = def.arena->allocatedBytes();
+            }
+
```

Comment:
Shouldn't this be controlled by `def.allocates_memory_in_arena`? I am not particularly experienced in these merging algorithms (but as a small PR, I thought it should be easy to review), so I am not sure, but it is suspicious to me.